### PR TITLE
Add `mdtop`

### PR DESCRIPTION
**Pull request by SimonBoothroyd**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by SimonBoothroyd** [comment]
@conda-forge/help-python, this should be ready for review!

File: recipes/mdtop/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/SimonBoothroyd/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: e35e594755d258d52da571925d076a221900c98243139fff005a89afb3fd4f19
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/mdtop/meta.yaml:14]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation 
```

File: recipes/mdtop/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  number: 0
+
+requirements:
+  build:
+    - python >=3.10
```
Comment (ocefpaf) [inline recipes/mdtop/meta.yaml:19]:
```suggestion
    - python {{ python_min }}
```

File: recipes/mdtop/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    - pip
+    - setuptools
+    - setuptools_scm >=8
+  run:
+    - python >=3.10
```
Comment (ocefpaf) [inline recipes/mdtop/meta.yaml:24]:
```suggestion
    - python >={{ python_min }}
```

File: recipes/mdtop/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    - pyparsing
+    - rdkit
+    - scipy
+    - openmm >=8.0.0
+
+test:
+  imports:
+    - mdtop
+  requires:
+    - pip
```
Comment (ocefpaf) [inline recipes/mdtop/meta.yaml:36]:
```suggestion
  requires:
    - python {{ python_min }}
    - pip
```

**Pull request merged**